### PR TITLE
Add new accent acrylic brushes

### LINKED ISSUE #4016: Add colored acrylic theme resources
# Add accent-colored acrylic theme resources

## Summary

Add new AcrylicBrush theme resources for showing accent-colored acrylics and overlays. Related to #3478. 

## Proposed changes

1. Add new theme resources

1.1 Create `AccentAcrylicBackgroundFillColorDefaultBrush` using the following properties.

  | TintColor | TintOpacity | TintLuminosityOpacity | FallbackColor | BackgroundSource
-- | -- | -- | -- | -- | --
Light theme | Accent Light 3 | 0.8 | 0.8 | Accent Light 3 | HostBackdrop
Dark theme | Accent Dark 1 | 0.8 | 0.8 | Accent Dark 1 | HostBackdrop

1.2 Create `AccentAcrylicBackgroundFillColorBaseBrush` using the following properties.

  | TintColor | TintOpacity | TintLuminosityOpacity | FallbackColor | BackgroundSource
-- | -- | -- | -- | -- | --
Light theme | Accent Light 2 | 0.8 | 0.8 | Accent Light 2 | HostBackdrop
Dark theme | Accent Dark 2 | 0.8 | 0.8 | Accent Dark 2 | HostBackdrop

1.3 Create a new SolidColorBrush named `AccentLayerFillColorDefaultBrush`

  | Fill color | Opacity
-- | -- | --
Light theme | Accent Light 3 | 0.25
Dark theme | Accent Dark 1 | 0.25

## Down-level behavior
The updated values described above rely on the TintLuminocityOpacity property, which was added in 19H1. Therefore, for the RS2 dictionary, we'll need to provide alternative values for these AcrylicBrush resources. At 80%, the TintOpacity should be enough to provide adequate contrast, so simply dropping the TintLuminocityOpacity values should work fine.

Proposed RS2 alternative values:
```xaml
<!-- Light theme -->
        <muxm:AcrylicBrush x:Key="AccentAcrylicBackgroundFillColorDefaultBrush" TintColor="{ThemeResource AccentLight3}" TintOpacity="0.8" FallbackColor="{ThemeResource AccentLight3}" BackgroundSource="HostBackdrop"/>
        <muxm:AcrylicBrush x:Key="AccentAcrylicBackgroundFillColorBaseBrush" TintColor="{ThemeResource AccentLight2}" TintOpacity="0.8" FallbackColor="{ThemeResource AccentLight2}" BackgroundSource="HostBackdrop"/>

<!-- Dark theme -->
        <muxm:AcrylicBrush x:Key="AccentAcrylicBackgroundFillColorDefaultBrush" TintColor="{ThemeResource AccentDark1}" TintOpacity="0.8" FallbackColor="{ThemeResource AccentDark1}" BackgroundSource="HostBackdrop"/>
        <muxm:AcrylicBrush x:Key="AccentAcrylicBackgroundFillColorBaseBrush" TintColor="{ThemeResource AccentDark2}" TintOpacity="0.8" FallbackColor="{ThemeResource AccentDark2}" BackgroundSource="HostBackdrop"/>
```

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

Adds new accent acrylic brush as per #4016
## Motivation and Context

Closes #4016
## How Has This Been Tested?

Tested manually (see screenshots)
## Screenshots (if appropriate):

New accent acrylic brushes in light theme

New accent acrylic brushes in dark theme

### CODE REVIEW

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml [hunk 1]
```diff
@@ -477,7 +488,17 @@
...
+                TintColor="{ThemeResource SystemAccentColorLight3}"
+                TintOpacity="0.8"
+                FallbackColor="{ThemeResource SystemAccentColorLight3}"
+                BackgroundSource="HostBackdrop"/>
+            <local:AcrylicBrush x:Key="AccentAcrylicBackgroundFillColorBaseBrush"
+                TintColor="{ThemeResource SystemAccentColorLight2}"
+                TintOpacity="0.8"
+                FallbackColor="{ThemeResource SystemAccentColorLight2}"
+                BackgroundSource="HostBackdrop"/>
+            <SolidColorBrush x:Key="AccentLayerFillColorDefaultBrush" Opacity="0.25" Color="{ThemeResource SystemAccentColorLight3}"/>
```
Comment (ranjeshj) [inline dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:501]: >AccentLayerFillColorDefaultBrush [](start = 36, length = 32)

Don't we need an entry for this brush in High Contrast ?

File: dev/Materials/Acrylic/TestUI/AcrylicColorPage.xaml.cs [hunk 2]
```diff
@@ -67,7 +67,8 @@ public static IEnumerable<string> AcrylicBrushNames
                     "SystemControlAltMediumLowAcrylicWindowMediumBrush",
                     "SystemControlAltMediumLowAcrylicElementMediumBrush",
                     "SystemControlAltHighAcrylicWindowBrush",
-                    "SystemControlAltHighAcrylicElementBrush" };
+                    "SystemControlAltHighAcrylicElementBrush"
+                };
```
Comment (ranjeshj) [inline dev/Materials/Acrylic/TestUI/AcrylicColorPage.xaml.cs:71]: What caused this test to fail ?

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml (hunk 1 above)
Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:501]: Yes, we do good point. Added it now.

File: dev/Materials/Acrylic/TestUI/AcrylicColorPage.xaml.cs (hunk 2 above)
Comment (chingucoding) [reply dev/Materials/Acrylic/TestUI/AcrylicColorPage.xaml.cs:71]: There is a test that uses a regex to validate the brushes names. The new brushes don't adhere to that pattern (e.g. don't start with `SystemControl`) so it failed the test. Not sure if keeping that test now makes sense though.

**Review by ranjeshj — APPROVED** [review]

File: dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml [hunk 3]
```diff
@@ -71,6 +91,8 @@
             <SolidColorBrush x:Key="AcrylicBackgroundFillColorDefaultInverseBrush" Color="{ThemeResource SystemColorWindowTextColor}" />
             <SolidColorBrush x:Key="AcrylicBackgroundFillColorBaseBrush" Color="{ThemeResource SystemColorWindowColor}"/>
             <SolidColorBrush x:Key="SystemControlDefaultBrighteningBrush" Color="{ThemeResource SystemColorWindowColor}"/>
+            <SolidColorBrush x:Key="AccentAcrylicBackgroundFillColorDefaultBrush" Color="{ThemeResource SystemAccentColor}"/>
+            <SolidColorBrush x:Key="AccentAcrylicBackgroundFillColorBaseBrush" Color="{ThemeResource SystemAccentColor}"/>
```
Comment (marksfoster) [inline dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml:95]: `<SolidColorBrush x:Key="AccentAcrylicBackgroundFillColorDefaultBrush" Color="{ThemeResource SystemColorWindowColor}"/>
            <SolidColorBrush x:Key="AccentAcrylicBackgroundFillColorBaseBrush" Color="{ThemeResource SystemColorWindowColor}"/>`

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml [hunk 4]
```diff
@@ -525,6 +546,9 @@
             <SolidColorBrush x:Key="AcrylicBackgroundFillColorDefaultInverseBrush" Color="{ThemeResource SystemColorWindowTextColor}" />
             <SolidColorBrush x:Key="AcrylicBackgroundFillColorBaseBrush" Color="{ThemeResource SystemColorWindowColor}"/>
             <SolidColorBrush x:Key="SystemControlDefaultBrighteningBrush" Color="{ThemeResource SystemColorWindowColor}"/>
+            <SolidColorBrush x:Key="AccentAcrylicBackgroundFillColorDefaultBrush" Color="{ThemeResource SystemAccentColor}"/>
+            <SolidColorBrush x:Key="AccentAcrylicBackgroundFillColorBaseBrush" Color="{ThemeResource SystemAccentColor}"/>
```
Comment (marksfoster) [inline dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:550]: `<SolidColorBrush x:Key="AccentAcrylicBackgroundFillColorDefaultBrush" Color="{ThemeResource SystemColorWindowColor}"/>
            <SolidColorBrush x:Key="AccentAcrylicBackgroundFillColorBaseBrush" Color="{ThemeResource SystemColorWindowColor}"/>`

File: dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml (hunk 3 above)
Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml:95]: Are those the correct high contrast colors to fallback to? @YuliKl can you weigh in?

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml [hunk 5]
```diff
@@ -525,6 +546,9 @@
...
+            <SolidColorBrush x:Key="AccentLayerFillColorDefaultBrush" Color="{ThemeResource SystemAccentColor}"/>
```
Comment (marksfoster) [inline dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:551]: `<SolidColorBrush x:Key="AccentLayerFillColorDefaultBrush" Color="{ThemeResource SystemColorWindowColor}"/>`

File: dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml (hunk 3 above)
Comment (YuliKl) [reply dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml:95]: Thanks for checking! Yes, the values @marksfoster listed for high contrast are correct. We actually don't want to show acrylic in HC at all.

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml [hunk 6]
```diff
@@ -240,6 +240,17 @@
...
+                TintColor="{ThemeResource SystemAccentColorDark1}"
+                TintOpacity="0.8"
+                FallbackColor="{ThemeResource SystemAccentColorDark1}"
+                BackgroundSource="HostBackdrop"/>
+            <local:AcrylicBrush x:Key="AccentAcrylicBackgroundFillColorBaseBrush"
+                TintColor="{ThemeResource SystemAccentColorDark2}"
+                TintOpacity="0.8"
+                FallbackColor="{ThemeResource SystemAccentColorDark2}"
+                BackgroundSource="HostBackdrop"/>
+            <SolidColorBrush x:Key="AccentLayerFillColorDefaultBrush" Opacity="0.25" Color="{ThemeResource SystemAccentColorDark1}"/>
```
Comment (marksfoster) [inline dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:253]: Move this brush to Common_themeresources.xaml below x:Key="LayerFillColorTertiaryBrush"

File: dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml (hunk 3 above)
Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml]: Alright, thanks for the clarification. Updated the HC brushes now.

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml (hunk 6 above)
Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:253]: Should we also move the lighting brush too? It's also not an acrylic brush.

Comment (marksfoster) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml]: Not sure what you mean by lighting brush

Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml]: Oh sorry, recalled the name incorrectly. I mean the `SystemControlDefaultBrighteningBrush` brush.

File: dev/CommonStyles/Common_themeresources.xaml [hunk 7]
```diff
@@ -333,6 +334,7 @@
             <Color x:Key="LayerFillColorDefault">#80FFFFFF</Color>
             <Color x:Key="LayerFillColorSecondary">#FFFFFF</Color>
             <Color x:Key="LayerFillColorTertiary">#40FFFFFF</Color>
+            <SolidColorBrush x:Key="AccentLayerFillColorDefaultBrush" Opacity="0.25" Color="{ThemeResource SystemAccentColorLight3}"/>
```
Comment (marksfoster) [inline dev/CommonStyles/Common_themeresources.xaml:337]: Move below x:Key="LayerFillColorTertiaryBrush"

Comment (chingucoding) [reply dev/CommonStyles/Common_themeresources.xaml]: Fixed now, thanks.

File: dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml [hunk 8]
```diff
@@ -35,6 +35,16 @@
                 FallbackColor="#202020"
                 BackgroundSource="HostBackdrop"/>
             <SolidColorBrush x:Key="SystemControlDefaultBrighteningBrush" Color="#FFFFFF" Opacity="0.0419"/>
```
Comment (marksfoster) [inline dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml:37]: You can remove this brush.

LayerFillColorDefaultBrush and AccentLayerFillColorDefaultBrush will be used instead for layers over acrylic.

@YuliKl for awareness.

Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml]: Okay, removed the brush now.

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml [hunk 9]
```diff
@@ -239,7 +239,16 @@
                 TintOpacity="0.8"
                 FallbackColor="#202020"
                 BackgroundSource="HostBackdrop"/>
```
Comment (marksfoster) [inline dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:241]: Can you update the non accent acrylic values to latest design POR in this PR or should I do it in a separate PR?

Here is the POR for dark theme:
image

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml [hunk 10]
```diff
@@ -476,8 +485,16 @@
                 TintOpacity="0.8"
                 FallbackColor="#F3F3F3"
                 BackgroundSource="HostBackdrop"/>
```
Comment (marksfoster) [inline dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:487]: Can you update the non accent acrylic values to latest design POR in this PR or should I do it in a separate PR?

Here is the POR for light theme:
image

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml (hunk 9 above)
Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:241]: Sure, I can update the values. Does "Luminosity" translate to the TintLuminosityOpacity property on AcrylicBrushes?

Comment (marksfoster) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:241]: Fantastic, thank you!

Yes:
Color % = TintOpacity
Luminocity % = TintLuminosityOpacity
Fallback hex = FallbackColor

Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:241]: Thanks, that helps a lot! For the RS2 versions, do we just leave out the luminosity opacity value or should we also adjust the opacity or color?

Comment (marksfoster) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:241]: I'm asking Yulia but it looks to me like for rs2 keep the existing TintOpacity (which will be different than the values in the POR and in 19h1) value but update the TintColor and FallbackColor to 19h1 values.

So I think the strategy is the following:

rs2:
TintColor  = POR value
FallbackColor = POR value
TintOpacity = current value in code

19h1:
TintColor  = POR value
FallbackColor = POR value
TintOpacity = POR value
TintLuminosityOpacity = POR value

Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:241]: Alright, updated the brushes now.

PS: There's a typo in the inverse brush element (Fallbacak).

File: dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml [hunk 11]
```diff
@@ -26,23 +26,32 @@
...
 
         <ResourceDictionary x:Key="Light">
             <local:AcrylicBrush
                 x:Key="AcrylicBackgroundFillColorDefaultBrush"
                 TintColor="#FCFCFC"
                 TintOpacity="0.0"
-                FallbackColor="#FCFCFC"
+                FallbackColor="#F9F9F9"
                 BackgroundSource="HostBackdrop"/>
             <local:AcrylicBrush
```
Comment (marksfoster) [inline dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml:56]: FallbackColor="#F9F9F9"

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml [hunk 12]
```diff
@@ -217,7 +217,7 @@
             <SolidColorBrush x:Key="SystemControlTransientBorderBrush" Color="#000000" Opacity="0.36" />
             <local:AcrylicBrush
                 x:Key="AcrylicBackgroundFillColorDefaultBrush"
-                TintColor="#1C1C1C"
+                TintColor="#2C2C2C"
                 TintOpacity="0.8"
                 FallbackColor="#2C2C2C"
                 BackgroundSource="HostBackdrop"/>
```
Comment (marksfoster) [inline dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:223]: TintColor="#2C2C2C" for AcrylicInAppFillColorDefaultBrush

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml [hunk 13]
```diff
@@ -456,7 +465,7 @@
                 x:Key="AcrylicBackgroundFillColorDefaultBrush"
                 TintColor="#FCFCFC"
                 TintOpacity="0.8"
-                FallbackColor="#FCFCFC"
+                FallbackColor="#F9F9F9"
                 BackgroundSource="HostBackdrop"/>
             <local:AcrylicBrush
                 x:Key="AcrylicInAppFillColorDefaultBrush"
```
Comment (marksfoster) [inline dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:471]: FallbackColor="#F9F9F9"

File: dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml (hunk 11 above)
Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml:56]: But the value already is #F9F9F9 😅

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml (hunk 12 above)
Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:223]: Updated now.

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml (hunk 13 above)
Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:471]: Same here, value is #F9F9F9.

File: dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml (hunk 11 above)
Comment (marksfoster) [reply dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml:56]: For AcrylicInAppFillColorDefaultBrush

AcrylicBackgroundFillColorDefaultBrush and AcrylicInAppFillColorDefaultBrush have identical values

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml (hunk 13 above)
Comment (marksfoster) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:471]: For AcrylicInAppFillColorDefaultBrush

AcrylicBackgroundFillColorDefaultBrush and AcrylicInAppFillColorDefaultBrush have identical values

File: dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml (hunk 11 above)
Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_19h1_themeresources.xaml:56]: Ah okay, updated now.

File: dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml (hunk 13 above)
Comment (chingucoding) [reply dev/Materials/Acrylic/AcrylicBrush_rs2_themeresources.xaml:471]: Fixed now.

**Review by marksfoster — APPROVED** [review]
